### PR TITLE
f-mega-modal@v5.0.0 - modal icing updates

### PR DESCRIPTION
**Pull request by jamieomaguire**
v5.0.0
------------------------------
*April 11, 2022*

### Added
- `isTextAlignedCenter` prop - this allows modal content to be center aligned (previously aligned center by default)
### Changed
- Modal content is now left aligned by default
- Modal title and close button now sit on the same line, and the close button sites on the baseline when the title breaks to multiple lines
- Close button inset positioning changed to stay in-line with the first line of title
- Increased modal widths by 10%
- Changed close button icon color to `$color-interactive-primary`

## Browsers Tested

- [X] Chrome (latest)
- [ ] Internet Explorer 11
- [ ] Mobile (Please list device/browser – Ideally one iPhone model and one Android)

### Screenshots
Default left alignment
Screenshot 2022-04-11 at 13 46 13

Really long modal title
Screenshot 2022-04-11 at 13 46 03

With `isTextAlignedCenter` set to true
Screenshot 2022-04-11 at 13 58 49

Flushed modal
Screenshot 2022-04-11 at 13 46 35

### CODE REVIEW

**Review by AlixRichards — APPROVED** [review]

**Review by dawidchar — APPROVED** [review]

File: packages/components/molecules/f-mega-modal/src/components/MegaModal.vue [hunk 1]
```diff
@@ -16,7 +16,8 @@
                 [$style['c-megaModal-content--wide']]: isWide,
                 [$style['c-megaModal-content--flush']]: isFlush,
                 [$style['is-fullHeight']]: isFullHeight,
-                [$style['is-positioned-bottom']]: isPositionedBottom
+                [$style['is-positioned-bottom']]: isPositionedBottom,
+                [$style['is-text-aligned-center']]: isTextAlignedCenter
```
Comment (ashleynolan) [inline packages/components/molecules/f-mega-modal/src/components/MegaModal.vue:20]: To fit naming, this should be `is-textAlignedCenter` (dashes should only be for indicating child hierarchy, camelCase is for just descriptive text.

If it's easy to do, the one above should also be `is-positionedBottom`

File: packages/components/molecules/f-mega-modal/src/components/MegaModal.vue [hunk 2]
```diff
@@ -367,32 +378,39 @@ export default {
...
-        padding: 0;
+.c-megaModal-content--flush {
+    padding-top: 0;
+    padding-bottom: 0;
+}
+
+.c-megaModal-content.c-megaModal-content--flush {
+    .c-megaModal-closeBtn {
+        top: spacing(d);
     }
```
Comment (ashleynolan) [inline packages/components/molecules/f-mega-modal/src/components/MegaModal.vue:412]: This should also work, just to group these a little clearer:

```scss
.c-megaModal-content--flush {
    padding-top: 0;
    padding-bottom: 0;

    &.c-megaModal-content {
        .c-megaModal-closeBtn {
            top: spacing(d);
        }
    }
}
```

**Review by ashleynolan — COMMENTED** [review]
Sorry for the delayed review – I was on leave most of this week.

Just a couple of things I spotted that would be good to update when you get chance.

File: packages/components/molecules/f-mega-modal/src/components/MegaModal.vue [hunk 3]
```diff
@@ -367,32 +378,39 @@ export default {
...
+    @include media('<mid') {
+        width: 100%;
+        max-width: 100%;
     }
+}
 
-    .c-megaModal-content--flush {
-        padding: 0;
+.c-megaModal-content--flush {
+    padding-top: 0;
```
Comment (DamianMullins) [inline packages/components/molecules/f-mega-modal/src/components/MegaModal.vue:405]: This change will break the modals in menu (and potentially in other projects) where we need to remove all spacing from the modal so we can handle it ourselves. Was there any particular reason for only applying to the top and bottom as it's no longer actually "flush"?

Comment (ashleynolan) [reply packages/components/molecules/f-mega-modal/src/components/MegaModal.vue:405]: @jamieomaguire Bump on this one

**Pull request merged**